idea/src/org/jetbrains/kotlin/idea/inspections/JavaCollectionsStaticMethodCallInspection.kt

```diff
@@ -0,0 +1,113 @@
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.idea.imports.importableFqName
+import org.jetbrains.kotlin.idea.intentions.callExpression
+import org.jetbrains.kotlin.js.descriptorUtils.nameIfStandardType
+import org.jetbrains.kotlin.load.java.descriptors.JavaMethodDescriptor
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.calls.callUtil.getResolvedCall
+import org.jetbrains.kotlin.resolve.calls.callUtil.getType
+import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
+
+class JavaCollectionsStaticMethodInspection : AbstractKotlinInspection() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : KtVisitorVoid() {
+            override fun visitDotQualifiedExpression(expression: KtDotQualifiedExpression) {
+                super.visitDotQualifiedExpression(expression)
+
+                val callExpression = expression.callExpression ?: return
+                val context = callExpression.analyze(BodyResolveMode.PARTIAL)
+                val resolvedCall = callExpression.getResolvedCall(context) ?: return
+                val descriptor = resolvedCall.resultingDescriptor as? JavaMethodDescriptor ?: return
+                val fqName = descriptor.importableFqName?.asString() ?: return
+                if (!canReplaceWithStdLib(fqName, callExpression, context)) return
+
+                holder.registerProblem(expression,
+                                       "Java Collections static method call can be replaced with Kotlin stdlib",
+                                       ProblemHighlightType.GENERIC_ERROR_OR_WARNING,
+                                       ReplaceWithSortFix(fqName))
+            }
+        }
+    }
+
+    private fun canReplaceWithStdLib(fqName: String, callExpression: KtCallExpression, context: BindingContext): Boolean {
+        if (!fqName.startsWith("java.util.Collections.")) return false
+        val valueArgs = callExpression.valueArguments
+        val valueArgsSize = valueArgs.size
+        if (valueArgsSize < 1) return false
+        if (!isMutableList(valueArgs[0], context)) return false
+        return when (fqName) {
+            "java.util.Collections.fill" -> valueArgsSize == 2
+            "java.util.Collections.reverse" -> valueArgsSize == 1
+            "java.util.Collections.shuffle" -> valueArgsSize == 1 || valueArgsSize == 2
+            "java.util.Collections.sort" -> valueArgsSize == 1 || valueArgsSize == 2
+            else -> false
+        }
+    }
+
+    private fun isMutableList(arg: KtValueArgument?, context: BindingContext): Boolean {
+        val expression = arg?.getArgumentExpression() ?: return false
+        return expression.getType(context)?.nameIfStandardType?.asString() == "MutableList"
+    }
+}
+
+private class ReplaceWithSortFix(private val fqName: String) : LocalQuickFix {
```

Comment:
The name of this class is no longer appropriate now that you've extended it to support other functions.